integration-tests/src/test/java/com/linkedin/kafka/clients/producer/LiKafkaProducerIntegrationTest.java

```diff
@@ -74,6 +78,12 @@ public void testSend() throws IOException, InterruptedException {
       while (messageCount < RECORD_COUNT && System.currentTimeMillis() < startMs + 30000) {
         ConsumerRecords<String, String> records = consumer.poll(100);
         for (ConsumerRecord<String, String> record : records) {
+          Map<String, byte[]> headers = LiKafkaClientsTestUtils.convertHeaders(record.headers());
+          assertTrue(headers.containsKey(Constants.TIMESTAMP));
+          assertTrue(
+              PrimitiveEncoderDecoder.decodeLong(headers.get(Constants.TIMESTAMP), 0)
+                  > System.currentTimeMillis() - TimeUnit.DAYS.toMillis(1)
```

Comment:
Can we record start and end timestamps of producer.send() above and make sure the timestamp in the header is within that range?